test/lib/rubycritic/generators/json_report_test.rb

```diff
@@ -0,0 +1,40 @@
+require 'test_helper'
+require 'rubycritic/core/analysed_modules_collection'
+require 'rubycritic/generators/json_report'
+require 'json'
+
+describe RubyCritic::Generator::JsonReport do
+  describe '#generate_report' do
+    before do
+      FileUtils.rm('test/samples/report.json') if File.file?('test/samples/report.json')
```

Comment:
Judging from experience I'd say that this will turn into a flickering test one day since you're operating on the real filesystem. How about using [fakefs](https://github.com/fakefs/fakefs) to defuse this bomb right from the start?